helm-charts/falcon-sensor/values.schema.json

```diff
@@ -521,41 +514,64 @@
     },
     "oneOf": [
         {
-            "properties": {
-                "falcon": {
-                    "properties": {
-                        "cid": {
-                            "type": "string",
-                            "minLength": 1
-                        }
-                    },
-                    "required": ["cid"]
-                }
-            },
-            "required": ["falcon"]
-        },
-        {
-            "properties": {
-                "falconSecret": {
-                    "properties": {
-                        "enabled": { "enum": [true] },
-                        "secretName": {
-                            "type": "string",
-                            "minLength": 1
+            "if": {
+                "properties": {
+                    "node": {
+                        "properties": {
+                            "cleanupOnly": {
+                                "type": ["null", "boolean"],
+                                "enum": [null, false]
+                            }
                         }
-                    },
-                    "required": ["enabled", "secretName"]
+                    }
                 }
             },
-            "required": ["falconSecret"],
-            "not": {
-                "properties": {
-                    "falcon": {
+            "then": {
+                "oneOf": [
+                    {
+                        "properties": {
+                            "falcon": {
+                                "properties": {
+                                    "cid": {
+                                        "type": "string",
+                                        "minLength": 1
+                                    }
+                                },
                         "required": ["cid"]
                     }
                 },
                 "required": ["falcon"]
-            }
+                    },
+                    {
+                        "properties": {
+                            "falconSecret": {
+                                "properties": {
+                                    "enabled": { "enum": [true] },
+                                    "secretName": {
+                                        "type": "string",
+                                        "minLength": 1,
+                                        "pattern": "^[0-9a-fA-F]{32}-[0-9a-fA-F]{2}$",
+                                        "example": [
+                                            "1234567890ABCDEF1234567890ABCDEF-12"
+                                        ]
```

Comment:
lines 553-556 are for `cid`, not `secretName`. It also looks like the indentation is not aligned properly on lines 540-543.